### PR TITLE
Expose Script component from next/script

### LINKED ISSUE #273: Allow custom JS code
The user should be able to give a string of custom JS code that will be compiled to the top of the page.

### LINKED ISSUE #357: Server side event to run Javascript and get return value
We are working on a new component to add customize script to compile output, but we also need a way to run Javascript code to access Cookie and LocalStorage ( in low-level, for more user friendly way, check #356 ).

Because run Javascript in client-side is not related to App state, we can place them in the package's namespace, for example, get the result of 1+1:

```python
result = pc.run_script("1 + 1")
```

Get Cookies string:

```python
cookies_string = pc.run_script("document.cookie")
```

### PR DESCRIPTION
**Pull request by masenf**
## Expose `Script`

The `Script` component allow inline or remotely loaded scripts as specified by the user. It provides event handlers that can call into the backend when the script is loaded/ready, or encounters an error. This component can be conditionally rendered to execute javascript in response to server side events. The `rx.client_side` special event is added to allow calling user-defined javascript functions in response to component events.

```python
import reflex as rx


class State(rx.State):
    """The app state."""
    should_log_to_console: bool = False

    def on_ready(self):
        print(f"READY (should_log_to_console={self.should_log_to_console})")


console_logging = """
function do_log(v) {
    console.log("hello javascript", v)
}"""


def index():
    return rx.fragment(
        rx.script(console_logging, on_ready=State.on_ready),
        rx.cond(
            State.should_log_to_console,
            rx.script("do_log('hey')"),
        ),
        rx.button("On", on_click=lambda: State.set_should_log_to_console(True)),
        rx.button("Off", on_click=lambda: State.set_should_log_to_console(False)),
        rx.button("Do it", on_click=rx.client_side("do_log(args)")),
    )


# Add state and page to the app.
app = rx.App(state=State)
app.add_page(index)
app.compile()
```

## [Rename reflex.components.base.document.Script as NextScript](https://github.com/reflex-dev/reflex/commit/b6950e8a6edb6e3a116c60a5251bb75d03575eb8)

`NextScript` is a special component that includes the scripts needed for nextjs itself. Rename accordingly to disambiguate the `Script` component from `next/script` (which facilitates execution of user supplied javascript).

This changes `rx.script` from `reflex.components.base.link.ScriptTag` to `reflex.components.base.script.Script` because the existing plain `ScriptTag` cannot usefully appear in a component definition since it is just a bare `<script>` tag and nextjs doesn't know how/when to render it.

address #1346 
fix #357 
fix #273

### CODE REVIEW

File: reflex/components/base/script.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    Note that this component differs from reflex.components.base.document.Script
+    in that it is intended for use with custom and user-defined scripts.
+    """
+
+    library = "next/script"
+    tag = "Script"
+    is_default = True
+
+    # Required unless inline script is used
+    src: Optional[Var[str]] = None
```
Comment (picklelo) [inline reflex/components/base/script.py:24]: We don't need to use `Optional` or set the default value to `None`, as that is done automatically for all props.

Can just do:
```python
src: Var[str]
strategy: Var[str]
...
```

File: reflex/components/base/script.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    def get_triggers(self) -> Set[str]:
+        """Get the event triggers for the component.
+
+        Returns:
+            The event triggers.
+        """
+        return super().get_triggers() | {"on_load", "on_ready", "on_error"}
+
+
+def client_side(javascript_code) -> Var[EventChain]:
```
Comment (picklelo) [inline reflex/components/base/script.py:68]: Nice! So if someone defines a `rx.script` with some functions, do you know if those functions would automatically become available here?

**Review by picklelo — COMMENTED** [review]
Looks good, just a comment on formatting the props

File: reflex/components/base/script.py (hunk 2 above)
Comment (masenf) [reply reflex/components/base/script.py:68]: Yes that's exactly right. Either in-line or in a separate js file, all defined functions declared with rx.script can be used with rx.client_side as event handlers.

File: reflex/components/base/script.py (hunk 1 above)
Comment (masenf) [reply reflex/components/base/script.py:24]: Okay thanks, good to know.

**Review by picklelo — APPROVED** [review]
looks good to me

**Pull request merged**